src/DiIiS-NA/D3-GameServer/GSSystem/PlayerSystem/ConversationManager.cs

```diff
@@ -100,20 +100,18 @@ private DiIiS_NA.Core.MPQ.FileFormats.Conversation asset
 
 		// Find a childnode with a matching class id, that one holds information about how long the speaker talks
 		// If there is no matching childnode, there must be one with -1 which only combines all class specific into one
-		private int duration
+		private int GetDuration()
 		{
-			get
+			var node = currentLineNode.ChildNodes.FirstOrDefault(a => a.ClassFilter == player.Toon.VoiceClassID);
+			node ??= currentLineNode.ChildNodes.FirstOrDefault(a => a.ClassFilter == -1);
+
+			if (node == null)
 			{
-				var node = from a in currentLineNode.ChildNodes
-					where a.ClassFilter == player.Toon.VoiceClassID
-					select a;
-				if (node.Count() == 0)
```

Comment:
Here we perform filtering by calling `Count()`, then again by calling the second `Count()` and again but not to the end when by calling `First()`.
Also I decided to convert this prop to method as it is pretty heavy for prop.